src/engine/bank02.asm

```diff
@@ -1,17 +1,1775 @@
+; the coordinates in screen are given by [hl]
+; input:
+; a  = tile for the icon
+; b  = value to print alongside icon
+; hl = pointer to coordinates
+DrawIconWithValue: ; 85e1 (2:45e1)
+; drawing the icon
+	ld d, [hl]
+	inc hl
+	ld e, [hl]
+	inc hl
+	push hl
+	push bc
+	lb hl, 1, 2
+	lb bc, 2, 2
+	call FillRectangle
+
+	ld a, [wConsole]
+	cp CONSOLE_CGB
+	jr nz, .skip
+
+	ld a, $02
+	lb bc, 2, 2
+	lb hl, 0, 0
+	call BankswitchVRAM1
+	call FillRectangle
+	call BankswitchVRAM0
+
+.skip
+; adjust coordinate to the lower right
+	inc d
+	inc d
+	inc e
+	call InitTextPrinting
+	pop bc
+	ld a, b
+	call CalculateOnesAndTensDigits
+
+	ld hl, wOnesAndTensPlace
+	ld a, [hli]
+	ld b, a
+	ld a, [hl]
+
+; loading numerical and cross symbols
+	ld hl, wDefaultText
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hl], SYM_CROSS
+	inc hl
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hli], a ; tens place
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld a, b
+	ld [hli], a ; ones place
+	ld [hl], TX_END
+
+; printing the decimal value
+	ld hl, wDefaultText
+	call ProcessText
+	pop hl
+	ret
+
+PlayAreaIconCoordinates ; 8635 (2:4635)
+; used for "Your/Opp. Play Area" screen
+.player1
+	db 15,  7 ; hand
+	db 15,  2 ; deck
+	db 15,  4 ; discard pile
+.opponent1
+	db  1,  5 ; hand
+	db  1,  9 ; deck
+	db  1,  7 ; discard pile
+
+; used for "In Play Area" screen
+.player2
+	db 15, 14
+	db 15,  9
+	db 15, 11
+.opponent2
+	db  0,  2
+	db  0,  6
+	db  0,  4
+
+; draws In Play Area icons
+DrawInPlayArea_Icons: ; 864d (2:464d)
+	ldh a, [hWhoseTurn]
+	ld d, a
+	ld e, DUELVARS_NUMBER_OF_CARDS_IN_HAND
+	ld a, [de]
+	ld b, a
+	ld a, $d0 ; hand icon, unused?
+	call PrintsHandTextAndValue
+
+; deck
+	ldh a, [hWhoseTurn]
+	ld d, a
+	ld e, DUELVARS_NUMBER_OF_CARDS_NOT_IN_DECK
+	ld a, [de]
+	ld b, a
+	ld a, DECK_SIZE
+	sub b
+	ld b, a
+	ld a, $d4 ; deck tile
+	call DrawIconWithValue
+
+; discard pile
+	ldh a, [hWhoseTurn]
+	ld d, a
+	ld e, $ed
+	ld a, [de]
+	ld b, a
+	ld a, $d8 ; discard pile tile
+	call DrawIconWithValue
+	ret
+
+; prints text HandText2 and a cross with 
+; decimal value of b
+; input
+; b = value to print alongside text
+PrintsHandTextAndValue: ; 8676 (2:4676)
+	ld d, [hl]
+	inc hl
+	ld e, [hl]
+	inc hl
+
+; text
+	push hl
+	push bc
+	call InitTextPrinting
+	ldtx hl, HandText2
+	call ProcessTextFromID
+	pop bc
+
+; decimal value
+	ld a, b
+	call CalculateOnesAndTensDigits
+	ld hl, wOnesAndTensPlace
+	ld a, [hli]
+	ld b, a
+	ld a, [hl]
+
+	ld hl, wDefaultText
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hl], SYM_CROSS
+	inc hl
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hli], a
+	ld [hl], TX_SYMBOL
+	inc hl
+
+; draw to screen
+	ld a, b
+	ld [hli], a
+	ld [hl], TX_END
+	ld hl, wDefaultText
+	call ProcessText
+	pop hl
+	ret
+
+; handle player input in menu in Opp. Play Area
+; works out which cursor coordinate to go to
+; and sets carry flag if A or B are pressed
+; input
+; returns a =  $1 if A pressed
+; returns a = $ff if B pressed
+HandleDuelMenuInput_PlayArea: ; 86ac (2:46ac)
```

Comment:
I think there's some confusion here but I don't blame you because so many menus feels like a rollercoaster :P
Anyway, this handles input in the "Your Play Area" and "Opp Play Area" submenus, so following your other conventions, it should probably be ``HandleDuelMenuInput_YourOrOppPlayArea``. (the comment header would also need an update)